File: enode/src/main/java/org/enodeframework/eventing/IProcessingEventProcessor.java
package org.enodeframework.eventing;

public interface IProcessingEventProcessor {
    /**
     * Process the given processingEvent.
     */
    void process(ProcessingEvent processingEvent);

    /**
     * Start the processor.
     */
    void start();

    /**
     * Stop the processor.
     */
    void stop();
}

File: enode/src/main/java/org/enodeframework/eventing/IProcessingEventProcessor.kt
package org.enodeframework.eventing

interface IProcessingEventProcessor {
    /**
     * Process the given processingEvent.
     */
    fun process(processingEvent: ProcessingEvent)

    /**
     * Start the processor.
     */
    fun start()

    /**
     * Stop the processor.
     */
    fun stop()

    /**
     * The name of the processor
     */
    val name: String
}